client/src/components/overview/Gallery.jsx

```diff
@@ -1,35 +1,122 @@
 import React, { useState, useEffect } from 'react';
 import GalleryThumbnails from './GalleryThumbnails.jsx';
+import arrowRight from '../../assets/arrow-right.svg';
+import arrowLeft from '../../assets/arrow-left.svg';
+import arrowUp from '../../assets/arrow-up.svg';
+import arrowDown from '../../assets/arrow-down.svg';
 
 const Gallery = ({ selectedStyle }) => {
 
-  const photos = selectedStyle.photos;
-
   // State and local variables
   // Strictly speaking, this is an anti-pattern. A piece of props is being saved to this component's state (mainImage). This led to a bug where the prop selectedStyle would change, but mainImage would not update. The ideal solution would be to store mainImage in the state of the parent component. I decided against that because the Gallery component is the only component that needs to use mainImage, and I didn't want to clutter the Overview component, which is already complex. The solution was to use the special key attribute on the invocation of the Gallery component in Overview. The style_id from selectedStyle is assigned to the key attribute, so when the selectedStyle changes, the key changes, and the Gallery component completely rerenders. Gallery's state is recomputed, and all is well. This solution would not be ideal if rerendering was expensive, but in this case, we will be rerendering most of the DOM elements anyways when selectedStyle changes.
+  const photos = selectedStyle.photos;
+  const THUMBNAIL_HEIGHT_PX = 60;
+  const THUMBNAIL_GUTTER_PX = 20;
+  const thumbnailsMaxScrollHeight =
+    THUMBNAIL_HEIGHT_PX * photos.length +
+    THUMBNAIL_GUTTER_PX * (photos.length - 1);
+  const thumbnailScrollIncrement =
+    THUMBNAIL_HEIGHT_PX * 7 +
+    THUMBNAIL_GUTTER_PX * 6;
+
   const [mainImage, setMainImage] = useState(photos[0]);
+  const [mainImageIndex, setMainImageIndex] = useState(0);
+  const [thumbnailScroll, setThumbnailScroll] = useState(0);
+
+  // Effects
+  useEffect(() => {
+    setMainImage(photos[mainImageIndex]);
+  }, [mainImageIndex]);
 
   // Event handlers
+  const handleThumbnailArrowClick = (event) => {
+    const direction = event.currentTarget.dataset.direction;
```

Comment:
This is new to me as well -- I thought you needed a `data-*` in front of your attributes to find in the `dataset`.